tools/dds/dds-server/dds-server.cpp

```diff
@@ -37,41 +65,41 @@ bool dds_server::init( DomainId_t domain_id )
 void dds_server::run()
 {
     _dds_device_dispatcher.start();
-    _running = true;
 
-    post_connected_devices_on_wakeup();
+    post_current_connected_devices();
 
     // Register to LRS device changes function to notify on future devices being
     // connected/disconnected
     _ctx.set_devices_changed_callback( [this]( rs2::event_information & info ) {
         if( _running )
         {
-            std::vector< std::string > devices_to_remove;
-            std::vector< std::pair< librealsense::dds::topics::device_info, rs2::device > > devices_to_add;
-
-            if( prepare_devices_changed_lists( info, devices_to_remove, devices_to_add ) )
-            {
-                // Post the devices connected / removed
-                _dds_device_dispatcher.invoke(
-                    [this, devices_to_remove, devices_to_add]( dispatcher::cancellable_timer c ) {
-                        post_device_changes( devices_to_remove, devices_to_add );
-                    } );
-            }
+            _dds_device_dispatcher.invoke( [this, info]( dispatcher::cancellable_timer ) {
+                std::vector< std::string > devices_to_remove;
+                std::vector< std::pair< std::string, rs2::device > > devices_to_add;
+
+                if( prepare_devices_changed_lists( info, devices_to_remove, devices_to_add ) )
+                {
+                    // Post the devices connected / removed
+                    handle_device_changes( devices_to_remove, devices_to_add );
+                }
+            } );
         }
     } );
-
+    
+    _new_client_handler.start();
+    _running = true;
     std::cout << "RS DDS Server is on.." << std::endl;
 }
 
 bool dds_server::prepare_devices_changed_lists(
     const rs2::event_information & info,
     std::vector< std::string > & devices_to_remove,
-    std::vector< std::pair< librealsense::dds::topics::device_info, rs2::device > > & devices_to_add )
+    std::vector< std::pair< std::string , rs2::device > > & devices_to_add )
 {
     // Remove disconnected devices from devices list
-    for( auto dev_info : _devices_writers )
+    for( auto dev_handle : _device_handle_by_sn )
     {
-        auto & dev = dev_info.second.device;
+        auto & dev = dev_handle.second.device;
         auto device_key = dev.get_info( RS2_CAMERA_INFO_SERIAL_NUMBER );
```

Comment:
Why, if we have the SN in the `sn_and_handle`?

Reply:
Correct, it's redundant